### ISSUE TITLE
Prediction - Storing Similarities

### ISSUE THREAD

**Issue by jacobmunson** [jacobmunson/RecommenderSystems#12]
When making predictions, let's try storing the calculated similarities  as they are computed, so that they don't need to be computed again.  For example, if had to compute sim(user1, user5), then may also need to compute sim(user5, user1) again - so let's store it and see what happens.

**Comment by jacobmunson** [comment]
It's unclear if this the stored data could communicate across cores when running in parallel.  Do this for a single core run first, see what speedup is like - then extend.

**Comment by jacobmunson** [comment]
This is challenging because "user1, user2, sim(user1,user2)" is tricky to retrieve since order can be reversed.

**Comment by jacobmunson** [comment]
I have this working on a single core - multicore is doable regardless, but will likely be much easier if splitting train set by users, so that a user doesn't get send to multiple cores.

**Comment by jacobmunson** [comment]
No longer relevant.
